### PR TITLE
use org.scala-refactoring snapshots for 2.10 branch and upgrade a

### PR DESCRIPTION
**Pull request by fommil**
variety of middleware components

Note that since scala 2.9 support is being dropped by the scala refactoring team

 https://github.com/scala-ide/scala-refactoring/issues/50

we should probably consider making a final 2.9.2 release and then dropping support from master... focussing on 2.10 and 2.11.

### CODE REVIEW

File: project/build.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
-sbt.version=0.12.0
+sbt.version=0.12.4
```
Comment (rajish) [inline project/build.properties:1]: What kept you from updating to 0.13.2 ?

File: project/Build.scala [hunk 2]
```diff
@@ -53,7 +53,7 @@ object EnsimeBuild extends Build {
   val root = Path(".")
 
   val TwoNineVersion = "2.9.2"
-  val TwoTenVersion = "2.10.2"
+  val TwoTenVersion = "2.10.4"
   val supportedScalaVersions = Seq(TwoNineVersion, TwoTenVersion)
```
Comment (rajish) [inline project/Build.scala:57]:
```
+ val TwoElevenVersion ??
```

Comment (fommil) [inline project/Build.scala:57]: I don't think it's quite that simple :-D

A pre-requisite to supporting 2.11 in the same source tree is that we drop 2.9 support. Many of the APIs we are using are simply not source compatible from 2.9 to 2.11. If anything, I recommend having separate branches.

File: project/build.properties (hunk 1 above)
Comment (fommil) [inline project/build.properties:1]: I only wanted to do minor version updates. From experience 0.13 introduces incompatibilities.

I would also like to upgrade lucene as recent releases have lots of new features, but again there are incompatibilities that need more consideration.

**Pull request merged**